### ISSUE TITLE
the input type = file can't detect the file change?

### ISSUE THREAD

**Issue by clevertension** [tsaikd/gogstash#21]
macOS Sierra 10.12.1, i write something to the file, but it can't detect it, please test it

**Comment by clevertension** [comment]
I find the root cause of this problem, the file watcher is on dir, not the file itself, when the directory is huge, for example "/Users/your_name", and the watcher may be busy
[test](http://awesomescreenshot.com/06a66q6z8a)

so i change to file itself and test it successfully

**Comment by tsaikd** [comment]
Watch directory could prevent inode change issue (not sure but more conservative). It happen when file system move file and create new file for the origin filename.

**Comment by clevertension** [comment]
Yes, i agree with you, so can you test with it?, or we can output as a configure item, and let the user determine to watch the **file** or **directory**
